x-pack/plugins/dashboard_enhanced/public/services/drilldowns/actions/flyout_edit_drilldown/flyout_edit_drilldown.tsx

```diff
@@ -95,5 +95,19 @@ export class FlyoutEditDrilldownAction implements Action<EmbeddableContext> {
     core.application.currentAppId$.pipe(takeUntil(closed$), skip(1), take(1)).subscribe(() => {
       close();
     });
+
+    // Close flyout on dashboard switch to "view" mode.
+    embeddable
+      .getInput$()
+      .pipe(
+        takeUntil(closed$),
+        map((input) => input.viewMode),
+        distinctUntilChanged(),
+        filter((mode) => mode !== ViewMode.EDIT),
```

Comment:
No need to change, but thought I'd share a nice trick: 

```
filter((mode) => mode !== ViewMode.EDIT),
take(1)
```

I think could be replaced with: 

```
first((mode) => mode !== ViewMode.EDIT)
```

Reply:
Thanks, this is a nice operator to know. The only thing that concerns me, and I'm not sure I completely understand the docs, this operator emits error, if no value was emitted:

![image](https://user-images.githubusercontent.com/9773803/117120146-438c5480-ad93-11eb-994b-21345c921289.png)

Reply:
In this specific case `take(1)` can actually be removed, but I like to keep it for future proof.

Reply:
Cool, didn't know about that error difference!